src/main/java/org/example/api/SystemApi.java

```diff
@@ -0,0 +1,36 @@
+package org.example.api;
+import org.example.model.Directory;
+import org.example.model.MyFile;
+
+public class SystemApi {
+    private final IFileApi api = new FileService();
+
+    public MyFile addFile(Directory directory, MyFile myFile) {
```

Comment:
I made some modifications to keep the logic, I didn't change the return types, because if I return Strings or byte arrays from my methods in that case readFile method becomes useless, but we obviously need that, so for users there is one more step to get the content by using readFile method